keps/sig-network/4444-service-routing-preference/README.md

```diff
@@ -0,0 +1,1128 @@
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP proposes introducing a new field, `routingPreference`, to the
+Kubernetes Service spec. It will supersede the functionality currently provided
+by the `service.kubernetes.io/topology-mode` annotation and it’s precursor
+`topologyKeys` field (which has been deprecated since Kubernetes 1.21)
+
+## Motivation
+
+To be able to understand the motivations for introducing this new field, it’s
+important to understand the precursors that shaped this KEP.
+
+
+### Topology Keys
+
+* This early mechanism allowed users to specify ordered routing preferences (e.g.,
+prioritize local nodes, then zone, then anywhere). 
+
+* **Limitations:** However, it lacked flexibility for implementations to make smart
+  decisions on its own, like incorporating feedback-based routing optimizations
+  (e.g., avoiding overloaded endpoints). The name `topologyKeys` might be
+  perceived as overly restrictive, given the desire to incorporate non-topology
+  factors into routing decisions.
+
+### Topology Aware Routing and InternalTrafficPolicy
+
+TopologyAwareRouting together with InternalTrafficPolicy were meant to be the
+successors of `topologyKeys` and allow implementations to be more flexible.
+
+* TopologyAwareRouting: 
+  * Exposes the annotation service.kubernetes.io/topology-mode. When this
+    annotation is set to Auto, an implementation specific heuristic is used to
+    route the traffic. 
+  * **Goal:** The aim with Auto was to allow implementations to be as smart as
+    possible and choose what it considered to be the best routing criteria. The
+    field didn’t support any other value by design with the thought that the
+    implementation should be able to make the best possible choice for the user.
```

Comment:
Maybe mention that this tried to find a balance between safety and same-zone routing, and unfortunately did not achieve either very well.